sched/scheduler/jobRunner.go

```diff
@@ -0,0 +1,267 @@
+package scheduler
+
+import (
+	"fmt"
+	"github.com/scootdev/scoot/saga"
+	"github.com/scootdev/scoot/sched"
+	cm "github.com/scootdev/scoot/sched/clustermembership"
+	dist "github.com/scootdev/scoot/sched/distributor"
+	"sync"
+)
+
+const MAX_RETRY = 3
```

Comment:
The retry is for writing to the saga log, right? (not rerunning the job if it fails)

I think we could leave this off here, and if we want retries, wrap the saga log in a RetryingSagaLog that retries as part of the saga log interface, which would make this code clearer because it's only doing one thing.